<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>626</width>
    <height>709</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="spacing">
        <number>6</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_17">
         <item>
          <widget class="QPushButton" name="pushButton">
           <property name="text">
            <string>测试用文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_Filename"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_18">
         <property name="spacing">
          <number>4</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>字节序(不影响RawData的读写)</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_19">
            <item>
             <widget class="QRadioButton" name="radioButton_BigEndian">
              <property name="text">
               <string>BigEndian</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButton_LittleEndian">
              <property name="text">
               <string>LittleEndian</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>浮点数精度(不影响RawData的读写)</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_20">
            <item>
             <widget class="QRadioButton" name="radioButton_DoublePrecision">
              <property name="text">
               <string>DoublePrecision</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButton_SinglePrecision">
              <property name="text">
               <string>SinglePrecision</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_ReadAndWriteDate">
      <property name="title">
       <string>原始二进制方式读写测试</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_16">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_15">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>9</number>
           </property>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>qint8(-128~127)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>quint8(0~255)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>qint16(-32768~32767)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>quint16(0~65535)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="layoutDirection">
              <enum>Qt::RightToLeft</enum>
             </property>
             <property name="text">
              <string>qint32</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>qint64</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Bool</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>qfloat16</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>float(4字节)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>double(8字节)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>String(UTF8编码)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>String(Latin1编码)</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>4</number>
           </property>
           <item>
            <widget class="QSpinBox" name="spinBox_Qint">
             <property name="minimum">
              <number>-128</number>
             </property>
             <property name="maximum">
              <number>127</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_QUint">
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Qint16">
             <property name="minimum">
              <number>-32768</number>
             </property>
             <property name="maximum">
              <number>32767</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_QUint16">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Qint32">
             <property name="minimum">
              <number>-999999999</number>
             </property>
             <property name="maximum">
              <number>2147483647</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_Qint64">
             <property name="minimum">
              <number>-999999999</number>
             </property>
             <property name="maximum">
              <number>999999999</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBox_Bool">
             <property name="text">
              <string>布尔值</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="spinBox_QFloat16">
             <property name="minimum">
              <double>-60000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>666666.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="spinBox_QFloat">
             <property name="minimum">
              <double>-9999999.000000000000000</double>
             </property>
             <property name="maximum">
              <double>99999999.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="spinBox_QDouble">
             <property name="minimum">
              <double>-999999999999999945575230987042816.000000000000000</double>
             </property>
             <property name="maximum">
              <double>999999999999999945575230987042816.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QCharString"/>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QString"/>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="pushButton_QInt8Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QInt8Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_Qint"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QPushButton" name="pushButton_QUint8Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QUint8Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QUint"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="pushButton_QInt16Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QInt16Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_Qint16"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="pushButton_QUint16Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QUint16Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QUint16"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="pushButton_QInt32Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QInt32Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_Qint32"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QPushButton" name="pushButton_QInt64Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QInt64Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_Qint64"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QPushButton" name="pushButton_BoolWrite">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_BoolRead">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBox_IsBool">
             <property name="text">
              <string>布尔值</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QPushButton" name="pushButton_QFloat16Write">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QFloat16Read">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QFloat16"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QPushButton" name="pushButton_QFloatWrite">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QFloatRead">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QFloat"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QPushButton" name="pushButton_QDoubleWrite">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QDoubleRead">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_QDouble"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QPushButton" name="pushButton_QCharWrite">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QCahrRead">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_ShowQCharString"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QPushButton" name="pushButton_QStringWrite">
             <property name="text">
              <string>写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QStringRead">
             <property name="text">
              <string>读出</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_ShowQString"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_13"/>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_14"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>626</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_EditClearAll"/>
   <addaction name="action_WriteToFile"/>
   <addaction name="action_ReadFromFile"/>
   <addaction name="separator"/>
   <addaction name="action_Quit"/>
  </widget>
  <action name="action_EditClearAll">
   <property name="text">
    <string>编缉框全清空</string>
   </property>
   <property name="toolTip">
    <string>清空全部编缉框</string>
   </property>
  </action>
  <action name="action_WriteToFile">
   <property name="text">
    <string>连续写入文件</string>
   </property>
   <property name="toolTip">
    <string>连续写入文件内容</string>
   </property>
  </action>
  <action name="action_ReadFromFile">
   <property name="text">
    <string>连续从文件读出</string>
   </property>
   <property name="toolTip">
    <string>连续从文件读出内容</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
